src/mavlink_stream.c

```diff
@@ -96,16 +102,16 @@ void mavlink_stream(uint32_t time_us)
   {
     if (mavlink_streams[i].period_us && time_us - mavlink_streams[i].last_time_us >= mavlink_streams[i].period_us)
     {
-      mavlink_streams[i].last_time_us = time_us;
+      // if we took too long, set the last_time_us to be where it should have been
+      mavlink_streams[i].last_time_us = time_us - (time_us - mavlink_streams[i].last_time_us-mavlink_streams[i].period_us);
       mavlink_streams[i].send_function();
     }
   }
 }
 
 void mavlink_stream_set_rate(mavlink_stream_id_t stream_id, uint32_t rate)
 {
-
-  mavlink_streams[stream_id].period_us = (rate == 0 ? 0 : 1e6 / rate);
+  mavlink_streams[stream_id].period_us = (rate == 0 ? 0 : (int32_t)((1000000.0f) / (float)rate));
```

Comment:
I think it is the same.  Again, I believe this float stuff is leftover from my other attempts at being smart about streaming.